vector/src/main/java/im/vector/listeners/ImageViewOnTouchListener.java

```diff
@@ -0,0 +1,123 @@
+package im.vector.listeners;
+
+import android.graphics.Matrix;
+import android.graphics.PointF;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.ImageView;
+
+public abstract class ImageViewOnTouchListener implements View.OnTouchListener {
```

Comment:
Where does this code comes from? Have you got a reference? If it has been copied from somewhre, please add ref.

Also please add some Javadoc to explain what this class is supposed to do

Reply:
Yeah that as from one tutorial and it worked fine. 
As i think about it a could ask him first :/
,but it was tutorial.
Where i have to link reference here:
[https://github.com/judepereira/android-multitouch ](https://github.com/judepereira/android-multitouch )
Or in java doc ?

Reply:
Please add the link in the Javadoc